pkg/middleware/namespace.go

```diff
@@ -0,0 +1,21 @@
+package middleware
+
+import (
+	"github.com/gofiber/fiber/v2"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func Namespace(c *fiber.Ctx) error {
+	namespace := c.Query("namespace", "default")
+
+	name := c.Params("name")
+
+	namespacedName := types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}
+	c.Locals("namespacedName", namespacedName)
+
+	c.Next()
+	return nil
+}
```

Comment:
can we only update locals with `namespace` to be used in next handler functions, or remove this middleware altogether in favor of
```
c.Query("namespace", "default")
```